### ISSUE TITLE
As an administrator, I want to upload a CSV to bulk invite users to join COs.

### ISSUE THREAD

**Issue by logan-life** [SURFscz/SBS#1676]
## Summary
Specific context: UU Yoda Data Managers currently use a bulk upload process for inviting users to join COs. The scale tends to be on the order of tens of users invited per week. 

Beyond the UU Yoda specific use case, in general it is a common workflow for people at the Data Manager / Unit Manager / Researcher Support level to work with CSVs for bulk operations.

## Acceptance Criteria
- [x] The user can be an Organization Administrator, Organization Manager, or Unit Manager.
- [x] The user can only successfully generate invitations for COs that fall under their purview (e.g. if the user is a Unit Manager, then the CO must fall within one of the user's Units)
- [x] The user is presented with a screen describing the required CSV schema and providing an example of a valid schema with dummy values.
- [x] The CSV file is validated for format and content when uploaded
- [x] The user is informed if the CSV validation was successful or unsuccessful.
- [x] If the CSV validation was **unsuccessful**, the user is informed as much as possible about why and where a problem occurred (Missing required value X on line Y, malformed value A on line B, etc)
- [x] If the CSV validation was **unsuccessful**, the invalid rows are filtered out when uploading the CSV
- [x] If the CSV validation was **successful**, the user is informed that the import succeeded 
- [x] After an upload with **successful** rows, the user is presented a `Proceed` function to actually upload the CSV to the server
- [x] Successful invitations are sent and use the existing invitation workflow for notifications
- [x] Unsuccessful invitations are reported back after the `Proceed`
- [x] Successful invitations can be viewed using the existing functionality within SBS (`Invited` status in the Members tab of a CO)
- [x] A user may re-import a previously imported CSV without any issues, in principle. For invitations sent via the prior import (or non-bulk process) that have not yet expired, the invitation should not be sent again and no error notification should be sent to the user. However the user is notified that duplicate invitations are not sent out. For invitations sent via the prior import (or non-bulk process) that have expired, new invitations are generated and sent per existing logic and workflow.

## Complex Cases
- [x] Because the CO and Group columns can have multiple values informing the user of Group 404 error for a given CO + Group combination may be a little tricky. Decided to not differentiate between success and failure within one row. Any invalid group will cause the entire row to be faulty.

## Nice To Have
- [x] More detailed feedback about a successful validation - show a table with the some information about what SBS understands from the import.
- [x] After being provided with the more detailed feedback after a successful validation, there is a "process" button that acts as a confirm step. Only after clicking the process button does SBS attempt to batch out the invitations.
- [ ] See screenshot for example UI for this part
![image](https://github.com/user-attachments/assets/38e8e9b3-d59f-48cd-b79b-1f2fc72b25f9)
- [x] Allow the user to provide the `membership_expiry_date` and `invitation_expiry_date` as ISO 8601 Date-Only values (e.g. `"2024-12-31"`) instead of `epoch_seconds`

## Notes
- [x] DELETE invitation is out of scope for bulk upload operations

## CSV Schema
- [x] 8 columns
- [x] Follows the conventions of the [Org API](https://sram.surf.nl/apidocs/#/Organisation/put_api_invitations_v1_collaboration_invites)
The first example shows single-values: 
    - Send one invitation to `rdoe@uniharderwijk.nl` to join the `cumulusgrp` CO with the `Admin` role and add them to the group ID `301ee8e6-b5d1-40b5-a27e-47611f803371`. The invitation expires at `epoch seconds` provided, and the user's membership in the CO expires at `epoch seconds` provided. The `message` and `sender_name` are included in the invitation.
- Second example shows multi-values:
    - Send four invitations -  `jdoe@surf.nl` + `cumulusgrp`, `jdoe@surf.nl` + `cirrusgrp`, `joost@surf.nl` + `cumulusgrp`, `joost@surf.nl` + `cirrusgrp`   and each with the `Member` role. Add the users to the group IDs supplied in the COs that the group belongs to (possible sticky point, see Complex Cases). The `invite_expiry` and `membership_expiry` are not provided, so use platform default logic. The `message` is provided. The `sender_name` is not provided, so use the default.
- Example CSV is attached: [bulk-import-example.csv](https://github.com/user-attachments/files/17661065/bulk-import-example.csv)
 

|`short_name(s)`|`intended_role`|`invitee(s)`|`group(s)`|`invitation_expiry_date`|`membership_expiry_date`|`message`|`sender_name`|
|:----------|:----------|:----------|:----------|:----------|:----------|:----------|:----------|
| string, required, multiple values allowed | string, required, [`admin`,`member`] | string (email address), required, multiple values allowed | string, optional, multiple values allowed | number, optional, defaults to current default logic | number, optional, default to current default logic | string, optional, no default | string, optional, default to current logic |
|----------|----------|----------|----------|----------|----------|----------|----------|
| cumulusgrp | admin | rdoe@uniharderwijk.nl | 301ee8e6-b5d1-40b5-a27e-47611f803371 | 1743014227174 | 1743014227174 | Please join the Cumulus research group collaboration page. | Organisation XYZ |
| cumulusgrp, cirrusgrp | member | jdoe@surf.nl, joost@surf.nl | 301ee8e6-b5d1-40b5-a27e-47611f803371, 412ee9e8-b6d2-51b6-a38e-58722f914482 | | | Please join our collaborations | |

**Comment by ccacciari** [comment]
![Image](https://github.com/user-attachments/assets/e72c52cf-ba17-4c78-86be-06002ec134b6)

**Comment by mrvanes** [comment]
I do not associate this with download:
![Image](https://github.com/user-attachments/assets/133de4a0-21a4-4341-869b-83d659b23625)

Should be "sent"
![Image](https://github.com/user-attachments/assets/6b0ac458-5b63-45ba-9c43-8799aeb01ee4)

As an uploader confronted with errors I would love to see real csv linenumbers in front of the error lines.
With real csv, I mean: 1 is the header, 2 is the first line containing invitations:
![Image](https://github.com/user-attachments/assets/2ca23d6c-8a1a-4ab8-876a-2049b3c162c5)

This line:
```
monitor1,,rdoe@uniharderwijk.nl, , , ,,Monitor1 Foobar
```
Results in a parsed error:
![Image](https://github.com/user-attachments/assets/c794dabb-b257-4c94-9951-8521d775807a)
The intended_role is correctly(?) defaulted to member, but then misses the intended invitee, which will result in deleting the outstanding invitation, that doesn't exist

`DELETE invitation is out of scope for bulk upload operations`
On the second upload, this line:
```
ai_computing,admin,rdoe@uniharderwijk.nl,4c270cff-de30-49e8-a3bc-df032536b37c,2025-10-12,2025-10-12,Please join the AI Computing and uuc_teacher research groups collaboration page.,Organisation XYZ
```
Created this report 
![Image](https://github.com/user-attachments/assets/a8ef5b10-ca04-43d6-80f5-b8914ff8ea8e)
Which is incorrect (the invitee was supplied) and threatens to remove the invitation, which it didn't?

**Comment by oharsta** [comment]
We can also silently ignore invitations, where there is an outstanding open invitation. But that also does not feel right.

**Comment by oharsta** [comment]
Did some rework. The feedback for duplicate / already outstanding emails looks like this.

<img width="1295" alt="Image" src="https://github.com/user-attachments/assets/b7210858-66c9-49a6-81a2-3259b44c5e36" />
